### PR TITLE
Added a questline for Sick Ampharos

### PR DESCRIPTION
**Pull request by RedzepR**
## Description

Added a questline for the sick ampharos story, including a new dungeon, NPCs and changing the requirements for Jasmines gym

## Motivation and Context

Wanted to add the lighthouse dungeon but it was kinda useless without the sick ampharos storyline so I've added it.
I'm open to any suggestion for improvement both regarding balance (of the dungeon) and the quest itself. Added stuff to the update.ts in hope that the game doesnt break 

## How Has This Been Tested?

Played through the quest with an old (past johto) and new (pre johto) save file

## Types of changes

- new questline

### CODE REVIEW

File: src/modules/GameConstants.ts [hunk 1]
```diff
@@ -1627,18 +1628,18 @@ export const HisuiDungeons = [
     'Ancient Lake Valor',
     'Ancient Lake Acuity',
     'Temple of Sinnoh',
-    'Turnback Cave', // 211
+    'Turnback Cave', // 212
 ];
 
 export const PaldeaDungeons = [
-    'Inlet Grotto', // 211
+    'Inlet Grotto', // 212
```
Comment (RegisCoaxans) [inline src/modules/GameConstants.ts:1635]: There was a misalignement. It should be 213.
Can you take advantage of the PR to fix this index and that of Area Zero Depths (220), please ?

File: src/scripts/quests/QuestLineHelper.ts [hunk 2]
```diff
@@ -390,6 +390,34 @@ class QuestLineHelper {
...
+        const talkToHerbalist1 = new TalkToNPCQuest(CianwoodCityPharmacist1, 'Talk to the Pharmacist in Cianwood City.');
+        sickAmpharosQuestLine.addQuest(talkToHerbalist1);
+
+        const clearCianwoodCityGym = new DefeatGymQuest(1, 0, 'Cianwood City').withDescription('The Pharmacist said he needs some time to finish Amphys medicine. Clear the Cianwood City Gym in the meantime.');
+        sickAmpharosQuestLine.addQuest(clearCianwoodCityGym);
+
+        const talkToHerbalist2 = new TalkToNPCQuest(CianwoodCityPharmacist2, 'Talk to the Pharmacist in Cianwood City.');
+        sickAmpharosQuestLine.addQuest(talkToHerbalist2);
+
+        const giveMedicineToAmphy = new TalkToNPCQuest(OlivineLighthouseMedicineAmphy, 'Give Amphy their medicine.');
```
Comment (Qwertypop04) [inline src/scripts/quests/QuestLineHelper.ts:412]: It's obvious, but just to be safe, could you specify that Amphy is in Olivine Lighthouse?

Comment (RedzepR) [reply src/scripts/quests/QuestLineHelper.ts]: Done

**Review by Qwertypop04 — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**